sklearn/feature_extraction/text.py

```diff
@@ -148,11 +160,21 @@ def _char_ngrams(self, text_document):
         text_document = self._white_spaces.sub(" ", text_document)
 
         text_len = len(text_document)
-        ngrams = []
         min_n, max_n = self.ngram_range
+        if min_n == 1:
+            # no need to do any slicing for unigrams
+            # iterate through the string
+            ngrams = list(text_document)
+            min_n += 1
+        else:
+            ngrams = []
+
+        # bind method outside of loop to reduce overhead
+        ngrams_append = ngrams.append
+
         for n in xrange(min_n, min(max_n + 1, text_len + 1)):
```

Comment:
Also benchmark returning a generator i.e. use `itertools.chain` over generator expressions to see if the counting can benefit from not materialising the list.